Custom Domain Names for Tenants

khorne3 opened the pull request:
Custom Domain Names, y'all:

* https://auth0-docs-content-pr-5216.herokuapp.com/docs/custom-domains
* CloudFront as proxy: https://auth0-docs-content-pr-5216.herokuapp.com/docs/custom-domains/set-up-cloudfront
* Azure as proxy: https://auth0-docs-content-pr-5216.herokuapp.com/docs/custom-domains/set-up-azure-cdn

Commit:
Initial commit

jhiner commented on custom-domains/cnames.md:
```diff
@@ -0,0 +1,72 @@
+
+You'll need to register and own the domain name to which you're mapping your Auth0 domain.
+
+## Sample Configuration
```
This isn't a "sample" is it? These are literally the steps you have to take.

jhiner commented on custom-domains/cnames.md:
```diff
@@ -0,0 +1,72 @@
+
+1. **If I use a custom domain, will I still be able to use my `tenant.auth0.com` domain to access Auth0?**
+
+	You can use `tenant.auth0.com` as long as you have *not* updated your hosted login page to support the custom domain.
```
You can use `tenant.auth0.com` **for centralized login**

jhiner commented on custom-domains/cnames.md:
```diff
@@ -0,0 +1,72 @@
+1. Provide your domain name to Auth0 and verify ownership
+1. Configure the reverse proxy
+1. Configure the hosted login page
+1. Set the domain as `active`
```
This should be _optional_: Enable custom domains in Auth0 emails

jhiner commented on custom-domains/cnames.md:
````diff
@@ -0,0 +1,72 @@
+    });
+```
+
+### Step 4: Set the Domain as `active`
````
Here we'll need to document the new switch being added to the Dashboard to enable custom domains in emails (https://github.com/auth0/manage/pull/3310/commits/05c820427e199524898032c0809635901b9994a6) 

This is an optional step.

jhiner commented on custom-domains/cnames.md:
```diff
@@ -0,0 +1,72 @@
+
+1. **Can I use my CNAME to access back-end flows, even if they aren't officially supported by Auth0?**
+
+	While your implementation may technically work, we strongly suggest that you use `tenant.auth0.com` for such instances since any other implementation isn't supported by Auth0.
```
Can I use my CNAME to access back-end flows? ~, even if they aren't officially supported by Auth0~

jhiner replied:
We only support using your `tenant.auth0.com` domain for back-end flows (e.g. `/oauth/token`).

jhiner reviewed (changes requested):
I left a few comments. I have not yet validated the cloudfront or azure steps though.

Commit:
Finish writing usage instructions

Commit:
Add screenshots for Azure

Commit:
Incorporate feedback

Commit:
Update and rename cnames.md to index.md

Commit:
Add toc

Commit:
Add headers and intro

Commit:
Add metadata and intro

Commit:
Compress and resize screenshots

khorne3 committed:
Move screenshots to media folder

khorne3 committed:
Remove dupes and update links

khorne3 committed:
Move docs under articles folder

khorne3 commented:
@jhiner First draft is done. I fixed the issues you mentioned, and added a bit more to the overview page. Azure and AWS proxy pages are also done as well.

I don't have screenshots for the dashboard -- is there a way for me to get them before you guys push to prod?

Commit:
Add requirement to Guardian

Commit:
Add screenshots

Commit:
Fix ordering

Commit:
Compress screenshots

jhiner commented on articles/custom-domains/index.md:
```diff
@@ -0,0 +1,104 @@
+
+## How to Configure Custom Domains
+
+Setting up your custom domain requires you to do the following steps:
```
Here the steps will differ if you are using Self-Managed Certs or Auth0-Managed Certs.

The steps you have here are fine for Self-Managed, but for Auth0-Managed you must:
- Create the custom domain
- In dashboard, you will be instructed to create the cname record in DNS
- Come back when that is done to verify (note it may take a few minutes or more to verify)
- It's ready.

jhiner commented on articles/custom-domains/index.md:
```diff
@@ -0,0 +1,104 @@
+
+You'll need to set up your reverse proxy. Currently, you can use [AWS CloudFront](/custom-domains/set-up-cloudfront) or [Azure CDN](/custom-domains/set-up-azure-cdn)
+
+### Step 3: Configure the Hosted Login Page
```
This step and step 4 are both common to both approaches.

Commit:
Update screenshots

Commit:
Resize screenshots

Commit:
Add notification

jhiner commented on articles/custom-domains/set-up-cloudfront.md:
```diff
@@ -0,0 +1,57 @@
+| Origin Domain Name | Set to `custom-domain-id.edge.tenants.auth0-region.auth0.com`. Be sure to replace the `custom-domain-id` and `auth0-region` placeholder values |
+| Origin ID | A description for the origin. This value lets you distinguish between multiple origins in the same distribution and therefore must be unique. |
+| Origin Protocol Policy | Set to **HTTPS Only** |
+
```
Is it possible to format this similar to how you have it in Azure CDN? 
https://github.com/auth0/docs/pull/5216/files#diff-12ece6d103347e7ccba7222486739c34R44

With the warning box about non-US regions and omit the auth0-region placeholder?

jhiner commented on articles/custom-domains/index.md:
```diff
@@ -0,0 +1,111 @@
+
+	You can use `tenant.auth0.com` for centralized login as long as you have *not* updated your hosted login page to support the custom domain.
+
+1. **Can I use my CNAME to access back-end flows?**
```
This may change and I will update the comments here if it does.

Commit:
Fix Cloudfront chart

jhiner commented on articles/custom-domains/index.md:
```diff
@@ -0,0 +1,111 @@
+
+When you've done so, click **Verify** to proceed.
+
+If Auth0 was able to verify your domain name, you'll see the following pop-up window. Be sure to save the information provided, especially the `cname-api-key` value, since this is the **only** time you'll see this value.
```
Can you please add the screenshot here of the popup with the cname-api-key?

<img width="681" alt="screen shot 2017-12-04 at 12 33 36 pm" src="https://user-images.githubusercontent.com/184385/33567331-c7638b46-d8f0-11e7-9667-ef18ac6ad4c3.png">

jhiner commented on articles/custom-domains/set-up-azure-cdn.md:
```diff
@@ -0,0 +1,144 @@
+
+You will need to [add your custom domain name to your Azure CDN endpoint](https://docs.microsoft.com/en-us/azure/cdn/cdn-map-content-to-custom-domain#register-a-custom-domain-for-an-azure-cdn-endpoint-using-the-intermediary-cdnverify-subdomain).
+
+However, before you proceed, create a CNAME entry in your DNS record for the custom domain name that points to the CDN endpoint name. It'll look something like this: `login.mydomain.com CNAME login-mydomain-com.azureedge.net`.
```
Creating a cname record from your custom domain to the Azure CDN is something we also need to include in the AWS CloudFront instructions (I added a separate note).

jhiner commented on articles/custom-domains/set-up-cloudfront.md:
```diff
@@ -0,0 +1,66 @@
+
+![](/media/articles/custom-domains/aws/distributions.png)
+
+Your CloudFront setup is now ready for use!
```
We need a new step: 

Add a new CNAME record to your DNS for your custom domain pointing to the CloudFront Domain Name for your Distribution (e.g. e2zwy42nt1feu7.cloudfront.net).

jhiner commented on articles/custom-domains/index.md:
```diff
@@ -0,0 +1,111 @@
+
+	![](/media/articles/custom-domains/auth0-managed.png)
+
+* **Self-Managed Certificates**: Add the TXT verification record listed in the Dashboard to your domain's DNS record.
```
Would it possible to add an "info box" or callout here that mentions basically:
- these steps vary by domain host provider, but generally: 
- Login to your domain management service (for example GoDaddy or Google Domains).
- Create a new record and add the verification:
- For the record type, choose TXT.
- In the Name field, enter your custom domain name (for example, for login.acme.com, enter login).
- In the Time to Live (TTL) field, you can leave the default.
- In the Value field, paste the verification value from the Dashboard.
- Save the record. The changes might take a few minutes to take effect.

Commit:
Incorporate feedback

Commit:
Add screenshot

jhiner committed:
Updates based on testing

jhiner commented:
FYI @khorne3  I hope you don't mind I made a few changes based on my latest round of testing.

khorne3 commented:
@jhiner Thanks for the heads-up. No, I don't mind at all.

jhiner committed:
Clarify support for cloudfront

jhiner committed:
Updates based on testing

jhiner committed:
Minor change

jhiner committed:
Specify supported connections

jhiner committed:
Minor updates

Commit:
Edit text.

Commit:
Flag --> Tag

Commit:
Delete set-up-azure-cdn.md
